### ISSUE TITLE
Problem while changing current figure size in Jupyter notebook

### ISSUE THREAD

**Issue by BlackboX2821** [matplotlib/matplotlib#11253]
### Bug report

**Bug summary**

In Jupyter Notebook, the problem occurs when dynamicaly changing the size of a ploted figure. The image is cropped and not totally displayed. (See code and example below)

**Code for reproduction**

1st cell > import declaration:

```python
%matplotlib notebook

import pandas as pd
import matplotlib.pyplot as plt
```
2nd cell > declaring dataframe and plot:

```python
df = pd.DataFrame([1,2,3,4,6])
df.plot()
```
<img width="391" alt="capture0" src="https://user-images.githubusercontent.com/26171096/40044446-a750d21c-5827-11e8-9da6-935308da213a.PNG">

3rd cell > changing current figsize:

```python
fig = plt.gcf()
fig.set_size_inches(5,5)
```
<img width="349" alt="capture1" src="https://user-images.githubusercontent.com/26171096/40044523-e1aa5da2-5827-11e8-8399-577b5faa866f.PNG">

As seen on the second screenshot the image is cropped and I had to manualy resize it for a good display.
 
**Note**

  - the problem doesn't occur when ploting and changing size are in the same notebook cell
  - the problem doesn't occur when the zoom lvl of my browser is between 50% 70%

**Matplotlib version**

  * Operating system: 'Windows 10 Profesionnel'  - v1709 - vsystem 16299.371
  * Browser : 'Firefox 59.0.3 (64 bits)'
  * Matplotlib version: '2.1.0'
  * Matplotlib backend (`print(matplotlib.get_backend())`): nbAgg
  * Python version: '3.6.3 |Anaconda, Inc.| (default, Oct 15 2017, 03:27:45) [MSC v.1900 64 bit (AMD64)]'
  * Jupyter version (if applicable): '4.4.0'
  * Other libraries: pandas version '0.20.3'

Installation by default with anaconda with version 'conda 4.3.30'

**Comment by avinashmnit30** [comment]
Use `fig.set_size_inches(5,5, forward = False) `.
`Forward` parameter is True by default.

**Comment by tacaswell** [comment]
So it _works_ if you do not forward the size?

Can you try with mpl 2.2.2?

**Comment by avinashmnit30** [comment]
I checked. Similar problem (with the same solution) exists in mpl 2.2.2 as well.

**Comment by BlackboX2821** [comment]
Well changing `forward` option to `false` isn't really correcting the problem. It's more a turnaround. It seems to disable the autoresize of the figure window. 

So if you want to scale down the figure it's ok, but if you want to scale it up you steel have a croped image. See example bellow.

```
%matplotlib notebook
import pandas as pd
import matplotlib.pyplot as plt

df = pd.DataFrame([1,2,3,4,6])
df.plot()
```

<img width="396" alt="matplotlib_resize" src="https://user-images.githubusercontent.com/26171096/40645102-9d0f9640-6325-11e8-8d72-e0ff32fdcec7.PNG">

Scale down exemple:

```
fig = plt.gcf()
fig.set_size_inches(5,5, forward = False)
```

<img width="413" alt="matplotlib_resize_55 " src="https://user-images.githubusercontent.com/26171096/40645256-1b86c160-6326-11e8-83ed-36a2a101389a.PNG">

Scale up exemple:

```
fig = plt.gcf()
fig.set_size_inches(8,8, forward = False)
```

<img width="432" alt="matplotlib_resize_88" src="https://user-images.githubusercontent.com/26171096/40645312-50b5aed2-6326-11e8-9ab5-04326aba2c9b.PNG">

 This behavior seems logical after all.

___

An other turn around solution could be to first display a big image then scale it down.

**Comment by ImportanceOfBeingErnest** [comment]
It would be a bit strange if using `forward=False` would solve this problem - after all you *want* to forward the figure size to take immediate effect. 

In any case, I cannot reproduce this problem with matplotlib 2.2.2. The figure size changes as expected.

**Comment by BlackboX2821** [comment]
Agreed with you @ImportanceOfBeingErnest 

Maybe the pb has been fixed in matplotlib 2.2.2. 
I'll try it but I have some computation for the next hours so i have to wait a bit...

**Comment by BlackboX2821** [comment]
Sorry i was absent for a week. 

The problem seems to be corected in matplotlib 2.2.2 i don't have this issue in version 2.2.2

Thanks every one for your participation :)
